connect/reports/renderers/json.py

```diff
@@ -39,3 +40,28 @@ def generate_report(self, data, output_file):
             with open(output_file, 'wb') as f:
                 f.write(orjson.dumps(data))
         return output_file
+
+    async def generate_report_async(self, data, output_file):
+        tokens = output_file.split('.')
+        if tokens[-1] != 'json':
+            output_file = f'{tokens[0]}.json'
+        if inspect.isgenerator(data):
```

Comment:
In this case, the if `inspect.isgenerator(data)` is not needed.
must be replaced with `if inspect.isasyncgen(data)`.